test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +602,392 @@ contract('KyberNetworkProxy', function(accounts) {
+            let networkAddr = await networkProxy.kyberNetwork();
+            assert(networkAddr == network.address, "missmatch network address");
+            let hintHandlerAddr = await networkProxy.hintHandler();
+            assert(hintHandlerAddr == matchingEngine.address, "missmatch network address");
+        });
+    });
+
+    describe("test events", async () => {
+        it("HintHandlerSet", async () => {
+            let newHintHandler = await MatchingEngine.new(admin);
+            let txResult = await networkProxy.setHintHandler(newHintHandler.address, { from: admin });
+            await expectEvent(txResult, "HintHandlerSet", {
+                hintHandler: newHintHandler.address
+            });
+            await networkProxy.setHintHandler(matchingEngine.address, { from: admin });
+        });
+
+        it("KyberNetworkSet", async () => {
+            let newNetWork = await KyberNetwork.new(admin);
+            let txResult = await networkProxy.setKyberNetwork(newNetWork.address, { from: admin });
+            await expectEvent(txResult, "KyberNetworkSet", {
+                newNetwork: newNetWork.address, oldNetwork: network.address,
+            })
+            await networkProxy.setKyberNetwork(network.address, { from: admin });
+        });
+
+        it("ExecuteTrade", async () => {
+            let srcToken = tokens[1]
+            let srcAmount = new BN(2).mul(new BN(10).pow(new BN(tokenDecimals[1])));
+            let destAddress = accounts[7];
+            let initialBalance = await Helper.getBalancePromise(destAddress);
+            let hint = web3.utils.fromAscii("PERM");
+            let fee = 5;
+
+            await srcToken.transfer(taker, srcAmount);
+            await srcToken.approve(networkProxy.address, srcAmount, { from: taker });
+            let rate = await networkProxy.getExpectedRateAfterFee(srcToken.address, ethAddress, srcAmount, fee, hint);
+            let txResult = await networkProxy.tradeWithHintAndFee(srcToken.address, srcAmount,
+                ethAddress, destAddress, maxDestAmt, calcMinRate(rate), platformWallet,
+                fee, hint, { from: taker }
+            );
+            let destBalanceAfter = await Helper.getBalancePromise(destAddress);
+            let destQty = new BN(destBalanceAfter).sub(new BN(initialBalance));
+            await expectEvent(txResult, "ExecuteTrade", {
+                trader: taker,
+                src: srcToken.address,
+                dest: ethAddress,
+                actualSrcAmount: srcAmount,
+                actualDestAmount: destQty,
+                platformWallet: platformWallet,
+                platformFeeBps: new BN(fee),
+            });
+        });
+    });
+
+    describe("test reverting when using contract zero address", async () => {
+        it("test set network to zero address", async () => {
+            await expectRevert.unspecified(networkProxy.setKyberNetwork(zeroAddress, { from: admin }));
```

Comment:
unspecified is used when no revert message is there.
but here we have a revert message.
and generally speaking.
if there is not revert message we should add one.

Reply:
updated in https://github.com/KyberNetwork/smart-contracts/pull/726/commits/e4d067f71b9400d43a678940a586bbb43991f02a